### PR TITLE
Added .editorconfig; added Javascript example to observer pattern

### PR DESCRIPTION
**Pull request by liamcurry**
Couple changes:
- Added an .editorconfig file to keep coding style consistent.
- Added UTF-8 meta tag.
- Removed some trailing whitespace.
- Made links stand out a little more on the "Patterns" page.
- Added a Javascript example to the observer pattern page, with slightly modified verbiage to match your ELI5 example. JS might be easier to grasp for newbies, and I think it'd be cool to have examples for multiple languages anyway. Kind of like [how this site does it](http://rosettacode.org/wiki/Category:Programming_Tasks).

[You can preview the changes here](http://liamcurry.com/DesignPatterns/).

[You can run the JS example here](http://jsbin.com/hagicuquru/1/edit?js,console).

If you like the JS example, I can add some for the other patterns as well. Let me know what you think.

### CODE REVIEW

**Comment by armstrhb** [comment]
Hey Bill,

Yeah these are great.  I wanted to put additional languages out there at a
future time (you're right; not everyone knows java), so including
javascript would be fine.  I can already see that we'll want to consider
having our code sections minimized by default.  Then, the user can click
and expand the one they wish to see.  Keeps out clutter.

Good stuff.

Also, I'm pretty pumped about hugo.  I read it a bit last night, and I'm
going to try and do this site with that.  Don't spend too much time on this
side of the project though, as I'll probably be looking at redoing this
site with hugo.  My biggest heartburn with doing github pages was that
there was no templating (that I knew of).  Thanks for sending that to me
early.

**Comment by liamcurry** [comment]
I dig this idea and had some free time today so I started converting things to Hugo. I'm taking a break for now, but I pushed my progress to [the `hugo` branch](https://github.com/liamcurry/DesignPatterns/tree/hugo). (see #2)
